modules/zone/variables.tf

```
@@ -1,10 +1,21 @@
 variable "name" {
   description = "Name of the zone to create"
   type        = string
+  default     = null
 }
 
 variable "tags" {
   description = "Map of tags to apply to the Route53 zone"
   type        = map(string)
   default     = {}
 }
+
+variable "vpc_id" {
+  type = object({
+    vpc_id = optional(string)
```

When specified, this should be required. I'd suggest changing the setup a bit here, and setting the default to null, and making this a required attribute (when the object is not null).

```suggestion
    vpc_id = string
```